src/routes/api/bookingRoute.js

```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+import BookingController from '../../controllers/BookingController';
+import verifyToken from '../../middlewares/verifyToken';
+import Conflict from '../../middlewares/Conflict';
+import isAccommodationAvail from '../../middlewares/isAccommodationAvail';
+import Validate from '../../middlewares/Validate';
+import checkInputDataError from '../../middlewares/checkInputDataError';
+import ValidateAccommodation from '../../middlewares/ValidateAccommodation';
+
+const { checkIfCheckInDateIsAsCheckOutDate } = ValidateAccommodation;
+const bookingRouter = express.Router();
+const { isTripFound, isRoomBooked } = Conflict;
+
+/**
+ * @swagger
+ *
+ * /trips/{tripId}/booking:
+ *    post:
+ *      summary: User can book an accommodation
+ *      tags: [Booking]
+*      parameters:
+ *        - name: tripId
+ *          in: path
+ *          description: Please enter the valid trip id
+ *          required: true
+ *          schema:
+ *            type: integer
+ *      requestBody:
+ *        required: true
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Booking'
+ *      responses:
+ *        "200":
+ *          description: A user schema
+ *
+ * components:
+ *    schemas:
+ *      Booking:
+ *        type: object
+ *        required:
+ *          - roomId
+ *          - checkInDate
+ *          - checkOutDate
+ *        properties:
+ *          roomId:
+ *            type: integer
+ *          checkInDate:
+ *            type: string
+ *          checkOutDate:
+ *            type: string
+ *        example:
+ *           roomId: 3
+ *           checkInDate: 2019-11-19
+ *           checkOutDate: 2019-11-19 11:11:41.668+02
+ */
+
+
+bookingRouter.post('/:tripId/booking', verifyToken, isTripFound, Validate.bookAccommodationRules(), checkInputDataError, checkIfCheckInDateIsAsCheckOutDate, isRoomBooked, isAccommodationAvail, BookingController.bookAccommodation);
+
```

Comment:
Would you mind adding isUserVerified middleware?
This will help us to protect our routes from the email which doesn't exist.

Reply:
yeah, Thanks for reminding me @victorkarangwa4 and It's now ok